### ISSUE TITLE
Npm compile doesn't work on Windows

### ISSUE THREAD

**Issue by cdupuis** [atomist/automation-client-ts#271]
Issue moved from atomist/github-sdm#327

Created by @johnsonr at 2018-04-11T06:04:23Z:

Failure at gql gen due to assuming Bash syntax

Probably in `automation-client` in `start.cli.ts`

**Comment by ddgenome** [comment]
@cdupuis any chance you could run this on MS Windows and paste the error? Perhaps the issue is that the Node.js executable on MS Windows is `node.exe` instead of `node`?

**Comment by cdupuis** [comment]
For sure. Will do.

**Comment by ddgenome** [comment]
When you download the Node.js MS Windows binary from https://nodejs.org/en/download/, it's name is `node.exe`.

**Comment by ddgenome** [comment]
It looks like scripts installed under the "bin" directive in the package.json may get a `.cmd` extension added to them on MS Windows.

**Comment by ddgenome** [comment]
@cdupuis you can also test branch `win-gql-gen-271` to see if that fixes it.

**Comment by jwalter** [comment]
The error (on my machine at least) is:

```
Running GraphQL code generator in 'C:\dev\src\atomist\github-sdm'
C:\dev\src\atomist\github-sdm\node_modules\.bin\gql-gen:2
basedir=$(dirname "$(echo "$0" | sed -e 's,\\,/,g')")
          ^^^^^^^

SyntaxError: missing ) after argument list
    at createScript (vm.js:80:10)
    at Object.runInThisContext (vm.js:139:10)
    at Module._compile (module.js:576:28)
    at Object.Module._extensions..js (module.js:623:10)
    at Module.load (module.js:531:32)
    at tryModuleLoad (module.js:494:12)
    at Function.Module._load (module.js:486:3)
    at Function.Module.runMain (module.js:653:10)
    at startup (bootstrap_node.js:187:16)
    at bootstrap_node.js:608:3
```

**Comment by ddgenome** [comment]
Thanks @jwalter! That's in the `gql-gen` scripts, which we do not have control over but we might be able to work around. Is there a `C:\dev\src\atomist\github-sdm\node_modules\.bin\gql-gen.cmd` file? If so, what are its contents?

**Comment by ddgenome** [comment]
Those lines do not appear in the `node_modules/.bin/gql-gen` script in my project. That error implies that on MS Windows, NPM does not symlink in files under `node_modules/.bin` like it does on POSIX-like systems, but installs two wrappers: one for Bourne shells and one for the DOS cmd.exe. I wonder what it does for .js files with a `#!/usr/bin/env node` first line. On POSIX-like systems, it makes them executable so they can be invoked directly. If it does something similarly smart on MS Windows, perhaps we should just leverage its cross-platform work instead of trying to roll our own.

**Comment by jwalter** [comment]
Yes there is:
```
@IF EXIST "%~dp0\node.exe" (
  "%~dp0\node.exe"  "%~dp0\..\graphql-code-generator\dist\index.js" %*
) ELSE (
  @SETLOCAL
  @SET PATHEXT=%PATHEXT:;.JS;=;%
  node  "%~dp0\..\graphql-code-generator\dist\index.js" %*
)
```
I'm a little curious, why are you running the shell script using node.js? Just invoking 'gql-gen' directly would work on Windows since the `.cmd` file takes precedence over the extension-less file.

**Comment by ddgenome** [comment]
Thanks again @jwalter. I am not sure of the origin of this code, although I have made modifications to it, but it appears that someone wanted these commands to run Node.js with optional command-line arguments supplied by the `ATOMIST_NODE_OPTIONS` environment variable. If that is no longer important, we might be able to just leverage the intelligence of NPM here.

**Comment by ddgenome** [comment]
@jessitron was it you that added the `ATOMIST_NODE_OPTIONS` stuff? Is it OK to remove it so that Windows builds work?

**Comment by jessitron** [comment]
I don't remember. It's fine with me to remove

Thank you for doing this.

**Comment by ddgenome** [comment]
Fix released in 0.13.1
